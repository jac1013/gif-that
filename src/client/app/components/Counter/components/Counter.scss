@import '../../Home/Home';

.counter-section {
  @extend .home-section;
  flex-direction: column;

  &__container {
    @extend .home-section;
    height: 0;

    &__number-display {
      font-size: 100px;
      font-weight: 500;
      line-height: 1.3;
      color: #fff;
      margin: 0 30px 0 30px;
    }

    &__button {
      display: flex;
      overflow: hidden;

      margin: 10px;
      padding: 12px 12px;

      cursor: pointer;
      user-select: none;
      transition: all 150ms linear;
      text-align: center;
      white-space: nowrap;
      text-decoration: none !important;
      outline: none;

      color: #fff;
      border: 0 none;
      border-radius: 50%;
      height: 100px;

      font-size: 50px;
      font-weight: 500;
      line-height: 1.3;

      -webkit-appearance: none;
      -moz-appearance:    none;
      appearance:         none;

      justify-content: center;
      align-items: center;
      flex: 0 0 100px;

      &:hover {
        transition: all 150ms linear;
        opacity: .85;
      }

      &:active {
        transition: all 150ms linear;
        opacity: .75;
      }

      &.-regular {
        background-color: #7D1935;

        &:hover {
          background-color: #9D1935;
          opacity: 1;
        }

        &:active {
          background-color: #d5d6d6;
          opacity: 1;
          color: #000;
        }
      }
    }
  }

  &__back-home {
    @extend .counter-section__container__number-display;
    padding: 0;
    font-size: 50px;
    margin-top: 150px;
  }
}
