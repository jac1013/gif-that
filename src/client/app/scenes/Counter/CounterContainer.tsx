import {increment, decrement} from '../../services/counter/actions';
import {connect} from 'react-redux';
import {Counter} from '../../components/Counter/components/Counter';
import {getCounter} from '../../services/counter/index';
import * as React from 'react';
import {CounterState} from '../../services/counter/reducers';

export interface CounterProps {
  increment: (event: React.MouseEvent<HTMLInputElement>) => void;
  decrement: (event: React.MouseEvent<HTMLInputElement>) => void;
}

export interface CounterViewProps extends CounterState, CounterProps {}

const mapStateToProps = (state: any) : CounterState => {
  return {
    number: getCounter(state)
  }
};

const mapDispatchToProps = (dispatch: any) : CounterProps => {
  return {
    increment: () => {
      dispatch(increment());
    },
    decrement: () => {
      dispatch(decrement());
    }
  }
};

export const CounterContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Counter);
