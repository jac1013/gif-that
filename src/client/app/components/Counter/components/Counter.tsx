import * as React from 'react';
import {CounterViewProps} from '../../../scenes/Counter/CounterContainer';

export const Counter = (props: CounterViewProps) => (
  <section>
    Hello World! {props.number}
    <button onClick={props.incrementHandler}>increment</button>
    <button onClick={props.decrementHandler}>decrement</button>
  </section>
);
