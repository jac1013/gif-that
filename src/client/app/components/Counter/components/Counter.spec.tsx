import {suite, test} from 'mocha-typescript';
import {expect} from 'chai';
import * as React from "react";
import * as renderer from 'react-test-renderer';
import {Counter} from './Counter';
import {ReactTestRenderer} from 'react-test-renderer';

@suite
class CounterTest {
  component: ReactTestRenderer;
  number: number = 0;
  before() {
    this.component = renderer.create(
      <Counter incrementHandler={() => {}} decrementHandler={() => {}} number={this.number}/>
    );
  }

  @test
  shouldRenderCorrectly() {
    expect(this.component.toJSON()).to.matchSnapshot();
  }

  @test
  shouldHaveCorrectNumber() {
    expect(this.component.toJSON().children).to.contain(this.number + '');
  }
}
