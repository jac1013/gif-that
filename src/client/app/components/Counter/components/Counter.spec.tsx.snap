// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterTest shouldRenderCorrectly 1`] = `
<section>
  Hello World! 
  0
  <button
    onClick={[Function]}
  >
    increment
  </button>
  <button
    onClick={[Function]}
  >
    decrement
  </button>
  We are done here, let's go to the 
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</section>
`;
